File: JAVA/MétodosdeConveniência.java
class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public double getPreco() {
        return preco;
    }

    public String getNome() {
        return nome;
    }

    public Produto somar(Produto outro) {
        String novoNome = this.nome + " & " + outro.nome;
        double novoPreco = this.preco + outro.preco;
        return new Produto(novoNome, novoPreco);
    }

    @Override
    public String toString() {
        return "Produto: " + nome + ", Preço: R$" + preco;
    }

    public static void main(String[] args) {
        Produto produto1 = new Produto("Notebook", 2500.00);
        Produto produto2 = new Produto("Mouse", 150.00);

        Produto produtoResultado = produto1.somar(produto2);
        System.out.println(produtoResultado);
    }
}

File: GOLANG/MétodosdeConveniência.go
package main

import "fmt"

type Produto struct {
    Nome  string
    Preco float64
}

func (p Produto) Somar(outro Produto) Produto {
    return Produto{
        Nome:  p.Nome + " & " + outro.Nome,
        Preco: p.Preco + outro.Preco,
    }
}

func main() {
    produto1 := Produto{"Notebook", 2500.00}
    produto2 := Produto{"Mouse", 150.00}

    produtoResultado := produto1.Somar(produto2)
    fmt.Printf("Produto resultante: %s, Preço total: R$%.2f\n", produtoResultado.Nome, produtoResultado.Preco)
}
